server/src/test/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStatsTests.java

```diff
@@ -232,13 +232,38 @@ public void testAccountsRegularIndices() {
             .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 4)
             .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
             .build();
-        IndexMetadata.Builder indexMetadata = new IndexMetadata.Builder("foo").settings(settings).putMapping(mapping);
-        Metadata metadata = new Metadata.Builder().put(indexMetadata).build();
-        AnalysisStats analysisStats = AnalysisStats.of(metadata, () -> {});
-        IndexFeatureStats expectedStats = new IndexFeatureStats("german");
-        expectedStats.count = 1;
-        expectedStats.indexCount = 1;
-        assertEquals(Collections.singleton(expectedStats), analysisStats.getUsedBuiltInAnalyzers());
+        Metadata metadata = new Metadata.Builder().put(new IndexMetadata.Builder("foo").settings(settings).putMapping(mapping)).build();
+        {
+            AnalysisStats analysisStats = AnalysisStats.of(metadata, () -> {});
+            IndexFeatureStats expectedStats = new IndexFeatureStats("german");
+            expectedStats.count = 1;
+            expectedStats.indexCount = 1;
+            assertEquals(Collections.singleton(expectedStats), analysisStats.getUsedBuiltInAnalyzers());
+        }
+
+        Metadata metadata2 = Metadata.builder(metadata)
+            .put(new IndexMetadata.Builder("bar").settings(settings).putMapping(mapping))
+            .build();
+        {
+            AnalysisStats analysisStats = AnalysisStats.of(metadata2, () -> {});
+            IndexFeatureStats expectedStats = new IndexFeatureStats("german");
+            expectedStats.count = 2;
+            expectedStats.indexCount = 2;
+            assertEquals(Collections.singleton(expectedStats), analysisStats.getUsedBuiltInAnalyzers());
+        }
+
+        Metadata metadata3 = Metadata.builder(metadata2).put(new IndexMetadata.Builder("baz").settings(settings).putMapping("""
+            {"properties":{"bar":{"type":"text","analyzer":"french"}}}""")).build();
+        {
+            AnalysisStats analysisStats = AnalysisStats.of(metadata3, () -> {});
+            IndexFeatureStats expectedStatsGerman = new IndexFeatureStats("german");
+            expectedStatsGerman.count = 2;
+            expectedStatsGerman.indexCount = 2;
+            IndexFeatureStats expectedStatsFrench = new IndexFeatureStats("french");
+            expectedStatsFrench.count = 1;
```

Comment:
Could we add maybe 3 french fields in the last example, to validate that we get `count=3` but `indexCount=1`?

Reply:
Sure adjusted :)